### PR TITLE
Share tekton-chains public key with all users

### PR DESCRIPTION
**Pull request by Roming22**
A secret is used as EnterpriseContract currently does not support ConfigMap.

Signed-off-by: Romain Arnaud <rarnaud@redhat.com>

### CODE REVIEW

**Review by gabemontero — APPROVED** [review]
works for me re: don't need shared config maps 

the rbac looks OK to me wrt granting sytem:authenticated, and I'm assuming needs necessitated restricting view access to the tekton-chains namespace, vs. exposing it to multiple namespaces (which is again another justification for not needing shared config maps)

File: operator/gitops/argocd/pipeline-service/tekton-chains/public-key.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
```
[the remark this answers is not in the document]
Comment (Roming22) [reply operator/gitops/argocd/pipeline-service/tekton-chains/public-key.yaml:3]: Using a regular `Role` ends with `Error from server (Forbidden): secrets "public-key" is forbidden: User "system:serviceaccount:test-tekton-chains:pipeline" cannot get resource "secrets" in API group "" in the namespace "tekton-chains": RBAC: clusterrole.rbac.authorization.k8s.io "tekton-chains-public-key-viewer" not found`.

My understanding is that a `ClusterRole` is required to access resources that do not belong to a user/serviceaccount original namespace.

**Review by gabemontero — APPROVED** [review]
/lgtm

File: operator/gitops/argocd/pipeline-service/tekton-chains/public-key.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (sbose78) [reply operator/gitops/argocd/pipeline-service/tekton-chains/public-key.yaml:3]: >  I was under the impression that you needed a ClusterRoleBinding as well for cross namespace access

Same here, pretty amazed it works as a RoleBinding.

**Review by sbose78 — APPROVED** [review]
Looks good!

**Pull request merged**